### DISCUSSION TITLE
How to delete a liveblocks room when all users have left?

### DISCUSSION THREAD

**Discussion by REDAMANS** [liveblocks/liveblocks#1912] [Help] [answered]
'm using liveblocks to create a multiplayer game, i want to delete the room after all players have left the game, i added a storage variable to check if a room is ready to be deleted (because when creating a room it will have no active users by default), so when at least one user joins it's ready to be deleted, i tried using beforeunload event :
```typescript
useEffect(() => {
    if(!readyToDelete || othersCount !== 0)return; // Do not run if there are other players
    const beforeUnload = async (e: BeforeUnloadEvent) => {
      e.preventDefault();
      await deleteRoom() // Deleting the room
    }
    window.addEventListener("beforeunload", beforeUnload);
    return () => {
      window.removeEventListener("beforeunload", beforeUnload);
    }
  }, [readyToDelete])
```
This works fine when i try to close the tab or the browser, or even when i change the url, but when i navigate back the event "beforeunload" isn't triggered, so the room will stay in liveblocks database, i tried looking for an event that handles navigating back or forward, but i cant find one, i also tried using useOthersListener, but like its name suggests, it only listens to others events, there's no self listener, my question is how can i achieve the behaviour i'm seeking ?

**Comment by nvie** [comment]
I believe browsers may deliberately not expose this event, to combat spammers that could otherwise open a popup to keep you on the current page, etc. I think only once you interact with the page first, before clicking the back button, this event would happen and be detectable. So maybe you're out of luck in this edge case.

Another option you can try as a fallback is to try and delete from from your backend, using our REST API:

- [Get Rooms API](https://liveblocks.io/docs/api-reference/rest-api-endpoints#Room)
- [Delete Room API](https://liveblocks.io/docs/api-reference/rest-api-endpoints#delete-rooms-roomId)

If you have a Node.js backend, you can use our `@liveblocks/node` package to do this, which exposes these equivalent methods:

- [`Liveblocks.getRooms()`](https://liveblocks.io/docs/api-reference/liveblocks-node#get-rooms)
- [`Liveblocks.deleteRoom()`](https://liveblocks.io/docs/api-reference/liveblocks-node#delete-rooms-roomId)

Hope that helps!

**Reply by REDAMANS** [reply, marked as answer]
Thank you, i solved the problem by using webhooks.
